packages/ra-ui-materialui/src/auth/LoginForm.tsx

```diff
@@ -77,30 +105,35 @@ const LoginForm = ({ classes, isLoading, handleSubmit, translate }) => (
         </CardActions>
     </form>
 );
-LoginForm.propTypes = {
-    ...propTypes,
-    classes: PropTypes.object,
-    redirectTo: PropTypes.string,
-};
 
-const mapStateToProps = state => ({ isLoading: state.admin.loading > 0 });
+const mapStateToProps = (state: ReduxState) => ({
+    isLoading: state.admin.loading > 0,
+});
 
-const enhance = compose(
+const enhance = compose<Props & EnhancedProps, Props>(
     withStyles(styles),
-    translate,
+    withTranslate,
     connect(mapStateToProps),
     reduxForm({
         form: 'signIn',
-        validate: (values, props) => {
-            const errors = {};
+        validate: (values: FormData, props: TranslationContextProps) => {
+            const errors = { username: '', password: '' };
```

Comment:
That means that both fields have an empty error, it's different from having no error.

Reply:
No, empty string is the same as no error